file/src/main/java/org/apache/calcite/adapter/file/CsvEnumerator.java

```diff
@@ -284,6 +361,17 @@ static CSVReader openCsv(Source source) throws IOException {
           return null;
         }
         return Double.parseDouble(string);
+      case DECIMAL:
+        if (string.length() == 0) {
+          return null;
+        }
+        BigDecimal result = new BigDecimal(string);
+        if (result.precision() > fieldType.getPrecision() || result.scale() != fieldType
+            .getScale()) {
+          throw new IllegalArgumentException(
```

Comment:
Does throwing make sense here? Or should we coerce?

Reply:
Also, best to use string format for this kind of thing as opposed to string concatenation.

Reply:
Done. Based coercion on PostgreSQL's implementation https://www.postgresql.org/docs/13/datatype-numeric.html#DATATYPE-NUMERIC-DECIMAL.

Reply:
Done. Based coercion on PostgreSQL's implementation https://www.postgresql.org/docs/13/datatype-numeric.html#DATATYPE-NUMERIC-DECIMAL.